syscall/devices/hostfs/hostfs.c

```diff
@@ -1132,6 +1132,11 @@ static struct oe_dirent* _hostfs_readdir(oe_fd_t* desc)
     if (retval != 0)
         OE_RAISE_ERRNO(OE_EINVAL);
 
+    /* Guard the special case that a host sets an arbitrarily value for
```

Comment:
LGTM. It will be good if we can call out that this behavior is OE-specific; i.e., OE expects the `d_reclen` to always be `sizeof(struct oe_dirent)` while this may be true in Linux (https://www.man7.org/linux/man-pages/man3/readdir.3.html).